src/components/structures/RoomDirectory.js

```diff
@@ -573,15 +521,24 @@ module.exports = React.createClass({
                         />
                         <NetworkDropdown config={this.props.config} protocols={this.protocols} onOptionChange={this.onOptionChange} />
                     </div>
-                    {content}
+                    { content }
                 </div>
             </div>
         );
-    }
+    },
 });
 
-// Similar to matrix-react-sdk's MatrixTools.getDisplayAliasForRoom
-// but works with the objects we get from the public room list
-function get_display_alias_for_room(room) {
-    return  room.canonical_alias || (room.aliases ? room.aliases[0] : "");
+function mapRoomDirectoryRoomObject(room) {
```

Comment:
Why are we creating a whole new type of object rather than just passing the room over? Having two different APIs flying around for the same data seems confusing at best, especially if it's just to convert snake_case to camelCase in the code.

Reply:
(this might be clearer if the method had a comment)

Reply:
Ths issue here is that publicRooms APIs return blindly with snake_case, whereas riot's classes/components (and js-sdk's Room class) uses camelCase